### PR TITLE
Reformatting the `--save_sections` output to have the format `<Scene>_<SecNum>_<SecName><extension>`

### LINKED ISSUE #3471: When rendering with option --save_section the rendered file names are appended with 0001, 0002, 0003 instead of the "name" attribute.
I am not sure if this shall be categorized as a bug, so let me know if I have to move it elsewhere.

As per title, when rendering with option `--save_section `the rendered file names are appended with 0001, 0002, 0003 instead of the "name" attribute.

for example, say that iI have something like the following:

```
class MyScene(Scene)
    def construct(self):
          self.next_section("MyFirstSection")
          # some code
          self.next_section("Section_Two")
          # some other code
```

and then I render with `--save-sections` option‚ then the rendered files have names:

```
MyScene_0001.mp4
MyScene_0002.mp4
```

whereas it would be nicer if they inherit the section name, e.g.

```
MyScene_MyFirstSection.mp4
MyScene_Section_Two.mp4
```

### PR DESCRIPTION
**Pull request by doaamuham**
## Overview: What does this pull request change?

Fixes the rendered file names to inherit section name. The PR has been created as part of a University course and it is not a bug but it helps the user by improving the expected output. This provides a solution to the issue opened. 

## Further Information and Comments
Fixes #3471 

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by chopan050** [comment]
Hello, thanks for your contribution!
I edited your post to link the affected issue.

We (the developer team) were discussing that it should, at least, be better to have **both** the number and the name.
The issue with getting rid of the section number is that the files become unorganized in the folder, which leads to issues for both the user and external software.

Maybe something like `<SCENE>_<section_number>_<section_name>.<extension>`?
It is important that the section number comes **before** the section name, to ensure a proper sorting of the section files.

**Comment by doaamuham** [comment]
Oh I see, yes I can work on modifying this to meet the requirements.

File: manim/scene/scene_file_writer.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ def next_section(self, name: str, type: str, skip_animations: bool) -> None:
             and not skip_animations
         ):
             # relative to index file
-            section_video = f"{self.output_name}_{len(self.sections):04}{config.movie_file_extension}"
+            section_video = f"{self.output_name}_{name}{config.movie_file_extension}"
```
Comment (MrDiver) [inline manim/scene/scene_file_writer.py:194]:
```suggestion
            section_video = f"{self.output_name}_{len(self.sections):04}_{name}{config.movie_file_extension}"
```

**Comment by MrDiver** [comment]
Also, the tests should be modified for the `--save_sections` because if the file names are changed then the tests need to be changed to reflect this behavior! It would be nice if you can also add this change!

**Comment by doaamuham** [comment]
Just modified the tests to account for the change and all tests pass! Feel free to merge my pull request.

**Review by MrDiver — APPROVED** [review]
Looks good!

**Comment by christopher-besch** [comment]
We should check that the Manim Editor is fine with this change.

**Pull request merged**